intelmq/bots/outputs/cif3/output.py

```diff
@@ -0,0 +1,237 @@
+
+        if not self.fireball:
+            self._submit_cif3_indicator(cif3_indicator)
+        elif len(self.indicator_list) > 0 and (
+            (
+                (datetime.now() - self.last_flushed).total_seconds() >
+                    self.indicator_list_max_seconds
+            ) or
+                len(self.indicator_list) >= self.indicator_list_max_records
+        ):
+            self._submit_cif3_indicator(self.indicator_list)
+            self.indicator_list.clear()
+        else:
+            if len(self.indicator_list) == 0:
+                self.last_flushed = datetime.now()
+            self.indicator_list.append(cif3_indicator)
+
+        self.acknowledge_message()
+
+    def _parse_event_to_cif3(self, intelmq_event):
+        """
+        Takes in an IntelMQ event, parses fields to those used by CIFv3
+        Returns CIFv3 Indicator object
+        """
+        # build new cif3 indicator dict
+        new_cif3_dict = {}
+        new_cif3_dict['tags'] = []
+
+        if 'feed.provider' in intelmq_event:
+            new_cif3_dict['provider'] = intelmq_event['feed.provider']
+        else:
+            new_cif3_dict['provider'] = 'IntelMQ'
+
+        # set the tags
+        if (self.add_feed_provider_as_tag and
+                'feed.provider' in intelmq_event):
+            new_tag = intelmq_event['feed.provider']
+            new_cif3_dict['tags'].append(new_tag)
+
+        matched_tag = False
+        if 'classification.type' in intelmq_event:
+            for classification in INTELMQ_CLASSIFICATION_TO_CIF_TAGS_MAP.keys():
+                if classification in intelmq_event['classification.type']:
+                    new_cif3_dict['tags'].extend(
+                        INTELMQ_CLASSIFICATION_TO_CIF_TAGS_MAP[classification].split(','))
+                    matched_tag = True
+            if not matched_tag:
+                new_cif3_dict['tags'].append(intelmq_event['classification.type'])
+
+        for new_tag in self.cif3_additional_tags:
+            new_cif3_dict['tags'].append(new_tag)
+
+        # map the confidence
+        if 'feed.accuracy' in intelmq_event:
+            if not self.cif3_static_confidence:
+                new_cif3_dict['confidence'] = (intelmq_event['feed.accuracy'] / 10)
+        if not new_cif3_dict.get('confidence'):
+            new_cif3_dict['confidence'] = self.cif3_feed_confidence
+
+        # map remaining IntelMQ fields to CIFv3 fields
+        for intelmq_type in INTELMQ_TO_CIF_FIELDS_MAP.keys():
+            if intelmq_type in intelmq_event:
+                cif3_field = INTELMQ_TO_CIF_FIELDS_MAP[intelmq_type]
+                new_cif3_dict[cif3_field] = intelmq_event[intelmq_type]
+
+        # build the CIFv3 indicator object from the dict
+        new_indicator = None
+        try:
+            new_indicator = Indicator(**new_cif3_dict)
+        except Exception as err:
+            self.logger.error(f"Error creating indicator: {err}")
```

Comment:
```suggestion
        except Exception as err:
            self.logger.error(f"Error creating indicator: {err}")
            raise
```
Otherwise `new_indicator` is not defined and we get an exception two lines below here